### ISSUE TITLE
NPE error using Block#setType

### ISSUE THREAD

**Issue by GC-spigot** [PaperMC/Paper#2897]
### What behaviour is expected:
block to be set

### What behaviour is observed:
thrown NPE error https://pastebin.com/raw/nFRf3QiX

### Steps/models to reproduce:
using this code
http://prntscr.com/qrygf7

### Plugin list:
Just my custom plugin

### Paper build number:
http://prntscr.com/qrygxl

**Comment by Trigary** [comment]
Unable to reproduce... That NPE means that there is no block registered in the block `IRegistry` for `Material.AIR`. which is so weird that I will stop looking at this right now.

**Comment by electronicboy** [comment]
Basically, need a minimal reproduction case, but, more than likely,
something is calling something at a stupid point in server initialization
and breaking the bukkit >nms registry

**Comment by GC-spigot** [comment]
I interact with the block, get it's type, location all fine. What I do is loop through 9x9x9 blocks and it breaks at some point, will try to create an universal way of reproducing it.

**Comment by Spottedleaf** [comment]
do you have api-version set in config?

**Comment by GC-spigot** [comment]
No, but assuming I'm setting the block to air it should not matter, does it? This does not happen on vanilla spigot, only when using paper so far

**Comment by Trigary** [comment]
Able to reproduce by not using `api-version: 1.13`.

**Comment by electronicboy** [comment]
```
[20:42:42] [Server thread/ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing 'rplblock' in pw.valaria.papertesty.Papertesty$3(rplblock)
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:156) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.dispatchCommand(CraftServer.java:711) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.PlayerConnection.handleCommand(PlayerConnection.java:1657) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:1497) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.PacketPlayInChat.a(PacketPlayInChat.java:47) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.PacketPlayInChat.a(PacketPlayInChat.java:1) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:895) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_231]
Caused by: java.lang.NullPointerException
	at org.bukkit.craftbukkit.v1_15_R1.block.data.CraftBlockData.newData(CraftBlockData.java:544) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.createBlockData(CraftServer.java:1858) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.createBlockData(CraftServer.java:1833) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at org.bukkit.Bukkit.createBlockData(Bukkit.java:1346) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at org.bukkit.Material.createBlockData(Material.java:3363) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at org.bukkit.craftbukkit.v1_15_R1.block.CraftBlock.setType(CraftBlock.java:164) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at org.bukkit.craftbukkit.v1_15_R1.block.CraftBlock.setType(CraftBlock.java:158) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	at pw.valaria.papertesty.Papertesty$3.execute(Papertesty.java:101) ~[?:?]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149) ~[spigot-1.15.2.jar:git-Spigot-800b93f-a0e88fb]
	... 16 more
```

Reproduced on spigot too

**Comment by electronicboy** [comment]
I'd suggest somebody go replicate this on spigot following their guidelines and create an issue on their tracker, should work in build 71 here, keeping this open until Spigot solves this...

**Comment by zachbr** [comment]
This was resolved when we pulled upstream commit b1e96bd5 in with 55b3a09.
